### PR TITLE
feat: link cryptobox xcframework instead of .a lib SQPIT-771

### PR DESCRIPTION
**Pull request by billypchan**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQPIT-771" title="SQPIT-771" target="_blank">TaskSQPIT-771</a>  Create XCFramework for cryptobox-ios
  </td></table>
  <br />
 

# What's new in this PR?

### Issues
We want to support testing with simulator running on M1 Mac to fix some x86 binary on M1 Mac issues. But cryptobox framework does not support arm64 simulator binary and traditional .a lib can not handle this case.

### Causes (Optional)

traditional .a lib can not hold two arm64 arch (device and simulator) in 1 .a file.

### Solutions

pack `cryptobox-ios` in XCFramework format.
replace the lib sodium header folder with the header folder from Carthage

### TODO
- [x] fix `testThatDecryptionWorks`
- [x] update wireapp libsodium repo and get XCFramework from there
- [x] move `cbox.h` inside cryptobox-ios XCFramework package? (instead of curl it in makefile)
- [ ] https://github.com/wireapp/wire-ios-utilities/pull/137

### CODE REVIEW

File: WireCryptobox.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -895,13 +895,15 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 54B5E1E11F0A482B001F43A6 /* project-debug.xcconfig */;
 			buildSettings = {
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
```
Comment (billypchan) [inline WireCryptobox.xcodeproj/project.pbxproj:898]:
```suggestion
```

File: WireCryptobox.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -895,13 +895,15 @@
...
 			};
 			name = Debug;
 		};
 		BA7EF9711B7109B600204A8E /* Release */ = {
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 54B5E1E21F0A482B001F43A6 /* project.xcconfig */;
 			buildSettings = {
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
```
Comment (billypchan) [inline WireCryptobox.xcodeproj/project.pbxproj:906]:
```suggestion
```

File: WireCryptoboxTests/ChaCha20StreamEncryptionTests.swift [hunk 3]
```diff
@@ -169,6 +169,7 @@ class ChaCha20StreamEncryptionTests: XCTestCase {
         let encryptedMessage =  Data(base64Encoded: "V0JVSQAAAT5xxW76YX91IgLvJwXeC5x+q/8To15mBzbsA6rc5Dzf7xRyWH+LYv+bscKxj3c7Fl7trr/9qt78lgA5ZtyjK7d2ZBdSYl4HLskPjyUIseTjAZjGKt+7MEXp8aVBey8ooGep")!
         
         // when
+        ///TODO: Bill on M1 mac this test failed for caught error: "decryptionFailed"
```
Comment (billypchan) [inline WireCryptoboxTests/ChaCha20StreamEncryptionTests.swift:172]:
```suggestion
```

File: Cartfile [hunk 4]
```diff
@@ -1 +1,3 @@
 github "wireapp/wire-ios-utilities" ~> 44.0.0
+github "wireapp/cryptobox-ios" "v1.1.0_xcframework"
+github "billypchan/libsodium" "1.0.14"
```
Comment (johnxnguyen) [inline Cartfile:3]: Should this be `wireapp/libsodium`?

Comment (billypchan) [reply Cartfile:3]: yes, I will update it after wireapp fork is updated

**Review by johnxnguyen — APPROVED** [review]

**Review by typfel — APPROVED** [review]

**Comment by billypchan** [comment]
release minor

**Comment by zenkins** [comment]
Hello, @billypchan! Merging PR for command **release minor** for PR #**79**. Will create a **minor** release after merged.
Next version is 31.2.0.

**Pull request merged**